### DISCUSSION TITLE
FV Vector Auxvars

### DISCUSSION THREAD

**Discussion by joe61vette** [idaholab/moose#16947] [Q&A General] [answered]
Hello:

I am trying to add a vector auxvar to an input file for a FV fluid problem.  The input is:

[AuxVariables]
  [porosity]
    order = CONSTANT
    family = MONOMIAL
    fv = true
    components = 3
    initial_condition = '1.0 1.0 1.0'
  []
[]

This fails with the message:

*** ERROR ***
Error creating ArrayMooseVariable name with base name porosity 

If I set fv to false, it works fine.  Am I doing something wrong?  If not, should I just go ahead and use it as an FEM auxvar as it is constant in time.

Thanks,
  Joe Kelly

**Comment by GiudGiud** [comment, marked as answer]
Hi Joe

I would recommend using a regular FV variable for porosity, as a single component should suffice. ArrayVariable support is increasing in FV, but is not ready yet. You could also specify 3 auxVariables to represent each component, as we do with u, v, w for the velocity variable in most of the tests in modules/navier_stokes/test/tests/finite_volume

FE (aux)variables, and FE array variables currently cannot be coupled to FV problems. This is also planned to be supported in the future. 

Best,
Guillaume

**Reply by joe61vette** [reply]
Thanks Guillaume:

For my application, I will have anisotropic porosity.  As I am just beginning the work on this, I can use one auxvar for now.  Thanks again for the help.

  Joe

**Reply by GiudGiud** [reply]
Ok we will make sure to support this.

In the 3 separate auxVariables approach, most (but likely not all, I will have to check the equations) kernels will be fine with accepting the relevant porosity component. In the kernel definition:
[kernel_name_x_component]
  type = PINSFVSomething
  porosity = porosity_x
[]
